go/vt/topo/test/testing.go

```diff
@@ -40,94 +40,108 @@ func newKeyRange(value string) *topodatapb.KeyRange {
 	return result
 }
 
+func executeTestSuite(f func(*testing.T, *topo.Server), t *testing.T, ts *topo.Server, ignoreList []string, name string) {
+	for _, n := range ignoreList {
+		if n == name {
+			return
+		}
+	}
+	f(t, ts)
+}
+
 // TopoServerTestSuite runs the full topo.Server/Conn test suite.
 // The factory method should return a topo.Server that has a single cell
 // called LocalCellName.
-func TopoServerTestSuite(t *testing.T, factory func() *topo.Server) {
+func TopoServerTestSuite(t *testing.T, factory func() *topo.Server, ignoreList []string) {
```

Comment:
What's the significance of `ignoreList`? I couldn't see where it is being used.

Reply:
Currently we have following implementation of Topo Servers

1. Zookeeper
2. Consul
3. Etcd
4. Memory
5. Tee
6. K8s
7. Fake

Out of them Zookeeper, Consul & Etcd have implemented TryLock with non-block implementation (in this PR). Rest of the topo servers have implemented TryLock as blocking call (i-e same as Lock operation).  
IgnoreList is used to ignore TryLock test case for topo servers which does not have non-blocking implementation of 'TryLock' yet. e.g go/vt/topo/memorytopo/server_test.go